snapcraft/tests/integration/snaps/version/snap/snapcraft.yaml

```diff
@@ -0,0 +1,9 @@
+name: version-test
+version: 2.10
+summary: test version string
+description: |
+  The version should be parsed as a string.
```

Comment:
There doesn't actually appear to be a test using that, is that true?

I agree with @sergiusens though, there are plenty of unit tests covering this stuff, might as well add one more case there.